src/registration/Register.tsx

```diff
@@ -0,0 +1,223 @@
+import _ from 'lodash/fp';
+import React, { ReactNode, useState } from 'react';
+import { refreshTerraProfile, signOut } from 'src/auth/auth';
+import { ButtonPrimary, ButtonSecondary, LabeledCheckbox } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import planet from 'src/images/register-planet.svg';
+import { Ajax } from 'src/libs/ajax';
+import { SamUserAttributes } from 'src/libs/ajax/User';
+import colors from 'src/libs/colors';
+import { reportError } from 'src/libs/error';
+import Events from 'src/libs/events';
+import { RegistrationLogo } from 'src/libs/logos';
+import { authStore, getTerraUser, TerraUser } from 'src/libs/state';
+import { CommunicationPreferencesCheckbox } from 'src/registration/CommunicationPreferencesCheckbox';
+import { InterestInTerraCheckbox } from 'src/registration/InterestInTerraCheckbox';
+import { LabelledTextInput } from 'src/registration/LabelledTextInput';
+import validate from 'validate.js';
+
+const constraints = (partOfOrg: boolean) => {
```

Comment:
boolean parameters are usually unclear to read. Suggest an options object instead.

```suggestion
const constraints = ({ partOfOrg }: { partOfOrg: boolean }) => {
```

https://alexkondov.com/should-you-pass-boolean-to-functions/ 
https://www.codereadability.com/what-are-javascript-options-objects/